dateparser/parser.py

```diff
@@ -81,6 +81,10 @@ class _time_parser(object):
         '%I %p',
         '%H:%M:%S.%f',
         '%I:%M:%S.%f %p',
+        '%-H:%-M',
+        '%-H:%-M:%-S',
+        '%-H:%-M %p',
+        '%-H:%-M:%-S %p'
```

Comment:
Woudn’t the fix for https://github.com/scrapinghub/dateparser/issues/571 be to simply add `'%H:%M %p'`? I’m worried that we are opening the door here for wrong interpretations, to parse as dates stuff that is not a date. None of the input strings from the test look to me like real-life date strings. I could understand the hours part having a single digit, and if that requires `%-H` then `%-H` should be OK, but `%-M` and `%-S` seem wrong to me, and I would rather keep them out of the default list.

Reply:
Yeah, That may be the case then we can simply add the single digit format for hours only as in '%-H:%M' etc.

Reply:
Hi @Gallaecio 
I needed some time to realize it, but it doesn't make any sense to add `'%-H:%M %p'` as is equivalent to `'%I:%M %p'` for values where hour < 10 and equivalent to `'%H:%M %p'` for values where hour >= 10

Anyway, I have just merged this: https://github.com/scrapinghub/dateparser/pull/634